CMakeLists.txt

```diff
@@ -218,6 +218,20 @@ option(QUDA_NUMA_NVML "experimental use of NVML to set numa affinity" OFF)
 option(QUDA_VERBOSE_BUILD "display kernel register usage" OFF)
 option(QUDA_BUILD_NATIVE_LAPACK "build the native blas/lapack library according to QUDA_TARGET" ON)
 
+option(QUDA_HETEROGENEOUS_ATOMIC "enable heterogeneous atomic support (CUDA >= 11.0)?" ON)
+if(QUDA_HETEROGENEOUS_ATOMIC AND CUDA_VERSION LESS 11.0)
+  message(
+  NOTICE
+  "Disabling heterogeneous atomic support (requires CUDA >= 11.0)")
+  set(QUDA_HETEROGENEOUS_ATOMIC OFF)
+endif()
+if(QUDA_HETEROGENEOUS_ATOMIC AND ${CMAKE_BUILD_TYPE} STREQUAL "SANITIZE")
+  message(
+  NOTICE
+  "Disabling heterogeneous atomic support (does not support SANITIZE build)")
```

Comment:
I think we should make this an error so there are no surprises.

Reply:
Fixed

Reply:
Not sure this is the best choice for Jenkins: 
```
- QUDA 1.0.0 (v0.9.0-3131-g0f8dc2108) **
-- cmake version: 3.15.7
-- Source location: <https://nvsocal2.caltech.edu:8443/job/qudabase_ubuntu1804_cuda102_cmake315/COMMS=-DQUDA_QMP=ON,GPUARCH=sm_70/ws/>
-- Build location: <https://nvsocal2.caltech.edu:8443/job/qudabase_ubuntu1804_cuda102_cmake315/COMMS=-DQUDA_QMP=ON,GPUARCH=sm_70/ws/native>
-- Build type: STRICT
-- QUDA target: CUDA
CMake Error at CMakeLists.txt:214 (message):
  QUDA_HETEROGENEOUS_ATOMIC=ON requires CUDA >= 11.0
```

Reply:
And probably this will cause a lot of user support for those still on CUDA < 11 as with the default config they will run into this error.